examples/fabric-admin/commands/pairing/DeviceSynchronization.cpp

```diff
@@ -0,0 +1,176 @@
+    VerifyOrDie(path.mClusterId == Clusters::BasicInformation::Id);
+
+    ChipLogProgress(NotSpecified, "Attribute data");
+    path.LogPath();
+
+    switch (path.mAttributeId)
+    {
+    case Clusters::BasicInformation::Attributes::UniqueID::Id:
+        mCurrentDeviceData.has_unique_id =
+            SuccessOrLog(data->GetString(mCurrentDeviceData.unique_id, sizeof(mCurrentDeviceData.unique_id)), "UniqueId");
+        break;
+    case Clusters::BasicInformation::Attributes::VendorName::Id:
+        mCurrentDeviceData.has_vendor_name =
+            SuccessOrLog(data->GetString(mCurrentDeviceData.vendor_name, sizeof(mCurrentDeviceData.vendor_name)), "VendorName");
+        break;
+    case Clusters::BasicInformation::Attributes::VendorID::Id:
+        mCurrentDeviceData.has_vendor_id = SuccessOrLog(data->Get(mCurrentDeviceData.vendor_id), "VendorID");
+        break;
+    case Clusters::BasicInformation::Attributes::ProductName::Id:
+        mCurrentDeviceData.has_product_name =
+            SuccessOrLog(data->GetString(mCurrentDeviceData.product_name, sizeof(mCurrentDeviceData.product_name)), "ProductName");
+        break;
+    case Clusters::BasicInformation::Attributes::ProductID::Id:
+        mCurrentDeviceData.has_product_id = SuccessOrLog(data->Get(mCurrentDeviceData.product_id), "ProductID");
+        break;
+    case Clusters::BasicInformation::Attributes::NodeLabel::Id:
+        mCurrentDeviceData.has_node_label =
+            SuccessOrLog(data->GetString(mCurrentDeviceData.node_label, sizeof(mCurrentDeviceData.node_label)), "NodeLabel");
+        break;
+    case Clusters::BasicInformation::Attributes::HardwareVersion::Id:
+        mCurrentDeviceData.has_hardware_version = SuccessOrLog(data->Get(mCurrentDeviceData.hardware_version), "HardwareVersion");
+        break;
+    case Clusters::BasicInformation::Attributes::HardwareVersionString::Id:
+        mCurrentDeviceData.has_hardware_version_string = SuccessOrLog(
+            data->GetString(mCurrentDeviceData.hardware_version_string, sizeof(mCurrentDeviceData.hardware_version_string)),
+            "HardwareVersionString");
+        break;
+    case Clusters::BasicInformation::Attributes::SoftwareVersion::Id:
+        mCurrentDeviceData.has_software_version = SuccessOrLog(data->Get(mCurrentDeviceData.software_version), "HardwareVersion");
+        break;
+    case Clusters::BasicInformation::Attributes::SoftwareVersionString::Id:
+        mCurrentDeviceData.has_software_version_string = SuccessOrLog(
+            data->GetString(mCurrentDeviceData.software_version_string, sizeof(mCurrentDeviceData.software_version_string)),
+            "SoftwareVersionString");
+        break;
+    default:
+        ChipLogProgress(NotSpecified, "Attribute data not processed");
+        break;
+    }
+}
+
+void DeviceSynchronizer::OnReportEnd()
+{
+    // Report end is at the end of all attributes (success)
+    AddSynchronizedDevice(mCurrentDeviceData);
```

Comment:
I think this is supposed to be in an `#if defined(PW_RPC_ENABLED)`

Reply:
Updated ... but what is the point then to have the fabric admin without RPC? I guess we could use CLI however it seems more reasonable to require this enabling...